src/components/enter-email/hooks.ts

```diff
@@ -0,0 +1,53 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+import { checkIsEmail } from 'src/utils/checkEmail';
+
+type ReturnType = {
+  email: string;
+  setEmail: Dispatch<SetStateAction<string>>;
+  isDisabled: boolean;
+  emailNotExists: boolean;
+  setEmailNotExists: Dispatch<SetStateAction<boolean>>;
+  checkEmail: (e: React.FormEvent<HTMLFormElement>) => void;
+  onChange: (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => void;
+  onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
+};
+
+export function useEnterEmail(): ReturnType {
+  const [email, setEmail] = useState('');
+  const isDisabled = email.trim().length < 1;
+  const [emailNotExists, setEmailNotExists] = useState(false);
```

Comment:
type.... also you don't have order in your component, why there is state, then constant, then state, you need to group similar types of code = constants to constants, states to states